### DISCUSSION TITLE
How to update task definition image tag

### DISCUSSION THREAD

**Discussion by leetmachine** [aws/aws-cdk#12717] [Q&A]
I'm converting an existing cloudformation to use CDK and curious about the correct workflow for updating our deployed application version (ie: the new ECR image tag). We do not deploy with the `latest` tag and instead rely on an appVersion number. Previously we used cloudformation parameters as a way for our Jenkins server to use the aws cli to call `update-stack-set` with the new appVersion parameter and the `--use-previous-template` flag. This made it easy for our Jenkins server to update the stacks, without being responsible for handling the cfn templates, and really, not even needing access to them. 

It also made it possible for me to deploy changes to the CFN without needing to specify a version number with `usePreviousValue=true` on the stackset update.

What is the suggested way to deploy a "parameter" style change in CDK? is it simply `cdk deploy myStack --parameters key=value`? Is there a way to do it without `cdk deploy`?

**Comment by tim-finnigan** [comment]
Hi @leetmachine thanks for your patience. Here is the CDK documentation on parameters: https://docs.aws.amazon.com/cdk/v2/guide/parameters.html. As referenced there, you can do `cdk deploy MyStack --parameters uploadBucketName=uploadbucket` like you mentioned above. But that documentation also notes:

>In general, we recommend against using AWS CloudFormation parameters with the AWS CDK. The usual ways to pass values into AWS CDK apps are [context values](https://docs.aws.amazon.com/cdk/v2/guide/context.html) and environment variables. Because they are not available at synthesis time, parameter values cannot be easily used for flow control and other purposes in your CDK app.
